cabal-install/Distribution/Client/Dependency/Modular/Linking.hs

```diff
@@ -401,7 +428,7 @@ verifyFlag' (FN (PI pn i) fn) lg = do
         vals  = map (`M.lookup` vsFlags vs) flags
     if allEqual (catMaybes vals) -- We ignore not-yet assigned flags
       then return ()
-      else conflict ( S.fromList (map F flags) `S.union` lgConflictSet lg
+      else conflict ( CS.fromList (map F flags) `CS.union` lgConflictSet lg
                     , "flag " ++ show fn ++ " incompatible"
                     )
```

Comment:
Yes, because `Q.A-flag` and `Q'.A-flag` are "direct dependencies" of `A` and therefore have to be linked. But you seem to be doing something different and assert that it's equivalent?